docs/content/tutorial/step_05.ngdoc

```diff
@@ -1,277 +1,168 @@
-service may have (services often depend upon other services).
-
-Note that the names of arguments are significant, because the injector uses these to look up the
-dependencies.
-
+We added a standard HTML `<input>` tag and used Angular's
+{@link ng.filter:filter filter} function to process the input for the
+{@link ng.directive:ngRepeat ngRepeat} directive.
 
-<img  class="diagram" src="img/tutorial/tutorial_05.png">
+By virtue of the {@link ng.directive:ngModel ngModel} directive, this lets a user enter search
+criteria and immediately see the effects of their search on the phone list. This new code
+demonstrates the following:
 
+* Data-binding: This is one of the core features in Angular. When the page loads, Angular binds the
+  value of the input box to the data model variable specified with `ngModel` and keeps the two in
+  sync.
 
-### `$` Prefix Naming Convention
+  In this code, the data that a user types into the input box (bound to **`$ctrl.query`**) is
+  immediately available as a filter input in the list repeater
+  (`phone in $ctrl.phones | filter:`**`$ctrl.query`**). When changes to the data model cause the
+  repeater's input to change, the repeater efficiently updates the DOM to reflect the current state
+  of the model.
 
-You can create your own services, and in fact we will do exactly that in step 11. As a naming
-convention, Angular's built-in services, Scope methods and a few other Angular APIs have a `$`
-prefix in front of the name.
+  <img  class="diagram" src="img/tutorial/tutorial_05.png">
 
-The `$` prefix is there to namespace Angular-provided services.
-To prevent collisions it's best to avoid naming your services and models anything that begins with a `$`.
+* Use of the `filter` filter: The {@link ng.filter:filter filter} function uses the `$ctrl.query`
+  value to create a new array that contains only those records that match the query.
 
-If you inspect a Scope, you may also notice some properties that begin with `$$`. These
-properties are considered private, and should not be accessed or modified.
+  `ngRepeat` automatically updates the view in response to the changing number of phones returned
+  by the `filter` filter. The process is completely transparent to the developer.
 
 
-### A Note on Minification
+# Testing
 
-Since Angular infers the controller's dependencies from the names of arguments to the controller's
-constructor function, if you were to [minify](http://goo.gl/SAnnsm) the JavaScript code for
-`PhoneListCtrl` controller, all of its function arguments would be minified as well, and the
-dependency injector would not be able to identify services correctly.
+In previous steps, we learned how to write and run unit tests. Unit tests are perfect for testing
+controllers and other parts of our application written in JavaScript, but they can't easily
+test DOM manipulation or the wiring of our application. For these, an end-to-end (E2E) test is a
```

Comment:
Good point. (Since `bindings` sounded more narrow-scoped than intended,) I changed it to: 

> can't easily test templates, DOM manipulation or interoperability of components and services

WDYT ?